packages/envied_generator/lib/src/generator.dart

```diff
@@ -118,7 +125,24 @@ final class EnviedGenerator extends GeneratorForAnnotation<Envied> {
 
     late final EnvVal? varValue;
 
-    if (envs.containsKey(varName)) {
+    if (environment) {
+      final String? envKey = envs[varName]?.raw;
+      if (envKey == null) {
+        throw InvalidGenerationSourceError(
+          'Expected to find an .env entry with a key of `$varName` for field `${field.name}` but none was found.',
+          element: field,
+        );
+      }
+      final String? env = Platform.environment[envKey];
+      if (env == null) {
+        throw InvalidGenerationSourceError(
+          'Expected to find an System environment variable named `$envKey` for field `${field.name}` but no value was found.',
+          element: field,
+        );
+      } else {
```

Comment:
The `else` here is unnecessary as the `throw` above will terminate execution. You can simply write
```dart
if (env == null) {
 throw InvalidGenerationSourceError(
   'Expected to find an System environment variable named `$envKey` for field `${field.name}` but no value was found.',
   element: field,
 );
} 

varValue = EnvVal(raw: env);
```

Reply:
Sure, I can drop the else.  It's just habit for me to always `if (x == null) { /* deal with null */ } else { /* dart knows it is definitely not null */ }`